### PR TITLE
RSDK-4027 - update doCommand to report when cartographer has processed all data

### PR DESCRIPTION
**Pull request by kim-mishra**
# Changes
* add capability to call `RunFinalOptimization` through the CAPI queue and relevant tests along the way
*  call `RunFinalOptimization` before reporting that cartographer has finished

### CODE REVIEW

File: viam-cartographer/src/carto_facade/carto_facade.cc [hunk 1]
```diff
@@ -1200,3 +1212,22 @@ extern int viam_carto_get_internal_state_response_destroy(
...
+        cf->RunFinalOptimization();
+    } catch (int err) {
+        return err;
+    } catch (std::exception &e) {
+        LOG(ERROR) << e.what();
+        return VIAM_CARTO_UNKNOWN_ERROR;
+    }
+
+    return VIAM_CARTO_SUCCESS;
+};
```
Comment (JohnN193) [inline viam-cartographer/src/carto_facade/carto_facade.cc:1233]: newline

File: sensorprocess/sensorprocess.go [hunk 2]
```diff
@@ -37,6 +38,10 @@ func Start(
 			return false
 		default:
 			if jobDone := addLidarReading(ctx, config); jobDone {
+				err := config.RunFinalOptimizationFunc(ctx, config.Timeout)
```
Comment (JohnN193) [inline sensorprocess/sensorprocess.go:41]: is there anything we need to add for other sensors? like if the lidar completes its own offline dataset before the imu, do we need to block the imu from trying to add sensors? @jeremyrhyde

**Review by JohnN193 — COMMENTED** [review]
going to test this in a CRJ first, will approve when thats done!

**Review by JohnN193 — APPROVED** [review]
LGTM, offline mode was successful in the CRJ. Cartographer also posts an info log when the optimization completes, which is a nice surprise. 

would still want some info to make sure this works fine with IMUs in offline mode. but integration tests are passing so maybe its okay?

File: sensorprocess/sensorprocess.go (hunk 2 above)
Comment (jeremyrhyde) [reply sensorprocess/sensorprocess.go:41]: I will add the additional blocking mechanisms i nthe IMU PR. Should be a simple:

```
if jobDone := addLidarReading(ctx, config); jobDone {
        stopped = true
        err := config.RunFinalOptimizationFunc(ctx, config.Timeout)
```

**Pull request merged**